### PR TITLE
Publish new artifact which shades the Scala Library 

### LINKED ISSUE #149: Publish additional artifact with shaded Scala Library
**Is your feature request related to a problem?**

* The FEEL Engine is compatible with a specific version of the Scala Library
* When my project already uses an incompatible version of the Scala Library, I cannot use the FEEL Engine in my project

**Describe the solution you'd like**

An additional artifact is published which shades the Scala Library.

**Related issues**

Camunda BPM: https://jira.camunda.com/browse/CAM-12136

### PR DESCRIPTION
**Pull request by tasso94**
## Description

The Scala Library is shaded inside a new artifact with the classifier `scala-shaded`

## Related issues

closes #149

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -235,6 +236,40 @@
...
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <shadedArtifactAttached>true</shadedArtifactAttached>
+              <shadedClassifierName>scala-shaded</shadedClassifierName>
+              <createDependencyReducedPom>true</createDependencyReducedPom>
+              <promoteTransitiveDependencies>false</promoteTransitiveDependencies>
+              <createSourcesJar>false</createSourcesJar>
```
Comment (tasso94) [inline pom.xml:254]:
```suggestion
              <createSourcesJar>true</createSourcesJar>
```
We should probably create a sources jar of the scala-shaded artifact.

**Review by saig0 — APPROVED** [review]
Looks good as far as I can tell :)

File: pom.xml (hunk 1 above)
Comment (koevskinikola) [reply pom.xml:254]: :+1: I think it would be good to have them as well.

**Review by koevskinikola — APPROVED** [review]
Looks good! You can flip the sources flag if you like. I think it would be useful to have them.

**Pull request merged**